lib/srv/regular/sshserver.go

```diff
@@ -1019,6 +1027,49 @@ func (s *Server) getServerResource() (types.Resource, error) {
 	return s.getServerInfo(), nil
 }
 
+func (s *Server) filterExistingNodes(instances server.EC2Instances) (server.EC2Instances, error) {
+	var filtered []*ec2.Instance
+	for _, inst := range instances.Instances {
+		instID := fmt.Sprintf("%s-%s",
+			instances.AccountID,
+			aws.StringValue(inst.InstanceId))
```

Comment:
I think we should be filtering by label, not node name which can get overridden.

The install script should make sure that the joining node includes AWS instance ID in a label e.g. `teleport.dev/instance-id: i-xyz123` and then we should filter by label here.